src/providers/emoji/Emojis.cpp

```diff
@@ -296,31 +319,61 @@ std::vector<boost::variant<EmotePtr, QString>> Emojis::parse(
 
         for (const std::shared_ptr<EmojiData> &emoji : possibleEmojis)
         {
+            int emojiNonQualifiedExtraCharacters =
+                emoji->nonQualified.length() - 1;
             int emojiExtraCharacters = emoji->value.length() - 1;
-            if (emojiExtraCharacters > remainingCharacters)
+            if (remainingCharacters >= emojiExtraCharacters)
             {
-                // It cannot be this emoji, there's not enough space for it
-                continue;
-            }
+                // look in emoji->value
+                bool match = true;
 
-            bool match = true;
+                for (int j = 1; j < emoji->value.length(); ++j)
+                {
+                    if (text.at(i + j) != emoji->value.at(j))
+                    {
+                        match = false;
 
-            for (int j = 1; j < emoji->value.length(); ++j)
-            {
-                if (text.at(i + j) != emoji->value.at(j))
+                        break;
+                    }
+                }
+
+                if (match)
                 {
-                    match = false;
+                    matchedEmoji = emoji;
+                    matchedEmojiLength = emoji->value.length();
 
                     break;
                 }
             }
-
-            if (match)
+            else if (remainingCharacters >= emojiNonQualifiedExtraCharacters)
             {
-                matchedEmoji = emoji;
-                matchedEmojiLength = emoji->value.length();
+                // This checking here relies on the fact that the nonQualified string
+                // always starts with the same byte as value (the unified string)
+                bool match = true;
+
+                for (int j = 1; j < emoji->nonQualified.length(); ++j)
+                {
+                    if (text.at(i + j) != emoji->nonQualified.at(j))
+                    {
+                        match = false;
+
+                        break;
+                    }
+                }
```

Comment:
To make this more readable, this can be written as 

```cpp
bool match = QStringView{emoji->nonQualified}.mid(1) == 
             QStringView{text}.mid(i + 1, emojiNonQualifiedExtraCharacters);
```

Though it should be noted that this will introduce a call to `QStringView operator==` which can't be inlined and this code is in a hot loop (up to about 1600 iterations per detected emoji).

Reply:
Working on benchmarks and then I'll look at this

Reply:
Benchmark showed the two implementations were equal, see [`92133d8` (#4840)](https://github.com/Chatterino/chatterino2/pull/4840/commits/92133d81daa4d15dcacd9915766e93bf44b1f01b)